astropy/units/tests/test_quantity.py

```diff
@@ -1055,12 +1055,31 @@ def test_quantity_pickelability():
     assert q1.unit == q2.unit
 
 
-def test_quantity_from_string():
-    with pytest.raises(TypeError):
-        q = u.Quantity(u.m * "5")
-        # the reverse should also fail once #1408 is in
+def test_quantity_initialisation_from_unit():
+    q = u.Quantity(u.m)
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('m')
```

Comment:
I'm a bit ambivalent about allowing a bare unit.  Somehow a Quantity is by definition something that has a value, and a Unit is just a bare unit.  But I don't feel very strongly about this and perhaps pragmatic beats purity.